### PR TITLE
Dao4

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: src/main/java/core/service/CarServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public void removeDriverFromCar(Driver driver, Car car) {
+        carDao.get(car.getId()).get().getDrivers().remove(driver);
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        return Storage.cars.stream()
+                .filter(x -> x.getDrivers()
+                        .stream()
+                        .map(d -> d.getId())
```
Comment (Max-Pochepets) [inline src/main/java/core/service/CarServiceImpl.java]:
```suggestion
                        .map(Driver::getId)
```

**Review by Max-Pochepets — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,3 @@
 .idea/*
 *.iml
-target/*git
\ No newline at end of file
+target/*
```
Comment (RomanDubovskyi) [inline .gitignore]: an empty line is missing at the end of the file

File: src/main/java/core/dao/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public Car update(Car car) {
+        Car old = get(car.getId()).get();
+        Storage.cars.set(Storage.cars.indexOf(old), car);
+        return old;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.cars.removeIf(x -> x.getId().equals(id));
```
Comment (RomanDubovskyi) [inline src/main/java/core/dao/CarDaoImpl.java]:
```suggestion
        return Storage.cars.removeIf(c -> c.getId().equals(id));
```
in streams usually the first letter of an element is being used (Car - c, Human - h, Book  - b)

File: src/main/java/core/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import core.db.Storage;
+import core.lib.Dao;
+import core.model.Manufacturer;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
```
Comment (RomanDubovskyi) [inline src/main/java/core/dao/ManufacturerDaoImpl.java]: this class should not be in the current PR, it's from prev HW

File: src/main/java/core/lib/Injector.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Injector {
```
Comment (RomanDubovskyi) [inline src/main/java/core/lib/Injector.java]: same here, remove old changes from the PR

File: src/main/java/core/service/CarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        carDao.get(car.getId()).get().getDrivers().add(driver);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        carDao.get(car.getId()).get().getDrivers().remove(driver);
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/service/CarServiceImpl.java:53]: let's move this logic to the dao layer, we should receive already filtered data from dao

File: src/main/java/core/service/CarServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        carDao.get(car.getId()).get().getDrivers().add(driver);
```
Comment (RomanDubovskyi) [inline src/main/java/core/service/CarServiceImpl.java]:
```suggestion
        car.getDrivers().add(driver);
        carDao.update(car);
```

File: src/main/java/core/service/CarServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/service/CarServiceImpl.java:47]: same approach sould be done here - > first we modify car object, then synchronize its' state in a db

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
remove classes from the previous hw from this PR

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**